.github/workflows/build-and-deploy-staging.yml

```diff
@@ -0,0 +1,146 @@
+name: Deploy Auctions UI staging
+
+on:
+    push:
+      branches: [ main ]
+
+jobs:
+  build-frontend:
+    name: build frontend
+    runs-on: ubuntu-latest
+    permissions: write-all
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Login to GitHub Packages
+      uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
+        username: ${{ secrets.GH_WORKFLOW_USER }}
+        password: ${{ secrets.GH_WORKFLOW_TOKEN }}
+
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        aws-region: eu-central-1
+        role-to-assume: arn:aws:iam::802385070966:role/GithubActionsSSMRole
+        role-session-name: AuctionsUiSession
+  
+    - name: Get secrets from parameterstore
+      uses: dkershner6/aws-ssm-getparameters-action@v2
+      with:
+        parameterPairs: "/auction-ui/main.auction-ui.k8s.sidestream.tech/frontend/rpc_url = RPC_URL, /auction-ui/main.auction-ui.k8s.sidestream.tech/frontend/contact_email = CONTACT_EMAIL, /auction-ui/main.auction-ui.k8s.sidestream.tech/frontend/frontend_origin = FRONTEND_ORIGIN, /auction-ui/main.auction-ui.k8s.sidestream.tech/frontend/infura_project_id = INFURA_POJECT_ID, /auction-ui/main.auction-ui.k8s.sidestream.tech/frontend/heapio_id = HEAPIO_ID, /auction-ui/main.auction-ui.k8s.sidestream.tech/frontend/production_domain = PRODUCTION_DOMAIN, /auction-ui/main.auction-ui.k8s.sidestream.tech/frontend/staging_banner_url = STAGING_BANNER_URL"
+        withDecryption: "true"
+
+    - name: Set outputs
+      id: vars
+      run: echo "git_hash_short=$(git rev-parse --short ${GITHUB_SHA})" >> $GITHUB_OUTPUT
+
+    - name: build and push frontend
+      uses: docker/build-push-action@v6
+      with:
+        file: frontend/Dockerfile
+        context: ./
+        platforms: linux/amd64
+        tags: |
+          ghcr.io/sidestream-tech/unified-auctions-ui/frontend:${{ github.sha }}
+        build-args: |
```

Comment:
As the Application DEMANDS (i have tried every other combination of run and build time environment parsing) the Envs as `build_args`, i added an extra step to pull the neccesary secrets from the SSM and parse them as `build_args` to the build step.